index.js

```diff
@@ -2,19 +2,516 @@ const readline = require('readline');
 const readlineInterface = readline.createInterface(process.stdin, process.stdout);
 
 function ask(questionText) {
-  return new Promise((resolve, reject) => {
-    readlineInterface.question(questionText, resolve);
-  });
+	return new Promise((resolve, reject) => {
+		readlineInterface.question(questionText, resolve);
+	});
 }
 
-start();
+let playerInventory = [];
+
+//assigning variables as undefined in order to make global functions out of the scope of the room's
+
+collectedItem = undefined;
+userAnswer = undefined;
+droppableItems = undefined;
+roomItems = undefined;
+
+/*-----------global take function that will remove item from the room's inventory and place it into the players inventory*/
+let take = function (userAnswer) {
+	for (let items of roomInv) {
+		collectedItem = items;
+	}
+
+	for (let items of playerInventory) {
+		droppableItems = items;
+	}
+	if (roomInv.length <= 0) {
+		console.log("There's nothing here...");
+	} else if (immovableItemsTable.includes(userAnswer)) {
+		console.log(`I cant take this...${userAnswer}`);
+	} else if (roomInv.length > 0 && collectedItem.item === userAnswer) {
+		roomInv.splice(roomInv.indexOf(collectedItem));
+		playerInventory.push(collectedItem);
+		console.log(`You collected a ${collectedItem.item}!`);
+	} else {
+		console.log("I don't know what you want...");
+	}
+};
+
+/*-------------global drop function that will remove item from players inventory and drop it into the rooms inventory-------------------*/
+let drop = function () {
+	for (let items of roomInv) {
+		collectedItem = items;
+	}
+
+	for (let items of playerInventory) {
+		droppableItems = items;
+	}
+	if (immovableItemsTable.includes(userAnswer)){
+    console.log(`No! i will not drop my precious ${userAnswer}`)
+  }else if(playerInventory.length > 0) {
+		droppableItems.item.includes(userAnswer);
+		playerInventory.splice(playerInventory.indexOf(droppableItems));
+		roomInv.push(droppableItems);
+		console.log(`You dropped a ${droppableItems.item}`);
+	} else {
+		console.log('I cant...');
+	}
+};
+
+//--------Global look around function to be called on in any room --------//
+let lookAround = function () {
+	if (roomInv.length !== 0) {
+		for (let roomItems of roomInv) {
+			collectedItem = roomItems.item;
+		}
+		console.log('\n' + roomDescription);
+		for (let roomItems of roomInv) {
+			console.log('you see a ' + roomItems.item);
+		}
+	} else if (roomInv.length === 0) {
+		console.log(roomDescription);
+		console.log("There's nothing here..");
+	}
+};
+
```

Comment:
I appreciate how to room inventory status changes once an item has been picked up already. This is nice functionality.